### ISSUE TITLE
Scalar product in objective

### ISSUE THREAD

**Issue by ohmsweetohm1** [JuliaOpt/JuMP.jl#2210]
```julia
l = ones(4)
m = JuMP.Model()
@variables(m, begin
    x
    y1
    y2
end)
epxr = [
    @expression(m, y1 - 1), 
    @expression(m, y2 - 2)]
```
**Does not work:**
```julia
@objective(m, Min,  x * 3 + l[2:3]' * epxr))
```
> MethodError: no method matching add_to_expression!(::GenericAffExpr{Float64,VariableRef}, ::Adjoint{Float64,Array{Float64,1}}, ::Array{GenericAffExpr{Float64,VariableRef},1})

**Works:**
```julia
@objective(m, Min,  l[2:3]' * epxr)) # without the term x*3
```
**Works too:**
```julia
exp2 = l[2:3]' * epxr
JuMP.@objective(model, Min, x * 3 + exp2)
```

**Comment by blegat** [comment]
Should be fixed by https://github.com/JuliaOpt/JuMP.jl/pull/2178
We should tag JuMP v0.21.2

**Comment by blegat** [comment]
Could you verify that updating to JuMP v0.21.2 fixes your issue ?

**Comment by odow** [comment]
Can confirm fixed by 0.21.2.
